codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/DefaultAuthSchemeProviderSpec.java

```diff
@@ -45,7 +45,10 @@ public DefaultAuthSchemeProviderSpec(IntermediateModel intermediateModel) {
 
     @Override
     public ClassName className() {
-        return authSchemeSpecUtils.providerDefaultImplName();
+        if (authSchemeSpecUtils.useEndpointBasedAuthProvider()) {
+            return authSchemeSpecUtils.internalModeledAuthSchemeProviderName();
+        }
+        return authSchemeSpecUtils.defaultAuthSchemeProviderName();
```

Comment:
with this, I think this class should be renamed to ModelBasedAuthSchemeProviderSpec (to go with EndpointBasedAuthSchemeProviderSpec).